apps/frontend/src/modules/Misc/components/global/Badge.vue

```diff
@@ -0,0 +1,25 @@
+<template>
+    <span
+        class="mr-2 rounded px-2.5 py-0.5 text-sm font-medium"
+        :class="{
+            'bg-green-100 text-green-800 dark:bg-green-200 dark:text-green-900': type === BadgeType.POSITIVE,
+            'bg-red-100 text-red-800 dark:bg-red-200 dark:text-red-900': type === BadgeType.NEGATIVE,
+            'bg-blue-100 text-blue-800 dark:bg-blue-200 dark:text-blue-900': type === BadgeType.NEUTRAL,
+        }"
+    >
+        {{ text }}</span
+    >
+</template>
+
+<script lang="ts" setup>
+    import { BadgeType } from '@/modules/Misc/enums/badge.enum';
+
+    interface IProps {
+        type: BadgeType;
+        text: string;
+    }
+
+    const { type, text } = defineProps<IProps>();
```

Comment:
Destructuring the props will cause the value to lose reactivity. Use props.type and props.text.